[skip ci] Update README.md with technical paper link (#105)
gear-tech/gear: README.md

## Getting Started

> Please refer to <a href="https://github.com/gear-tech/gear-technical/blob/master/TECHNICAL.pdf">technical paper</a> for some insights about how Gear works internally.

Reviewer: ```suggestion
Please refer to the [technical paper](https://github.com/gear-tech/gear-technical/blob/master/TECHNICAL.pdf) for some insights about how Gear works internally.
```
